pytoda/datasets/_csv_eager_dataset.py

```diff
@@ -26,23 +26,17 @@ def __init__(
             kwargs (dict): additional parameters for pd.read_csv.
                 Except from nrows.
         """
-        _CsvDataset.__init__(
+        _CsvStatistics.__init__(
             self, filepath, feature_list=feature_list, **kwargs
-        )
+        )  # calls setup_datasource
 
-    def setup_dataset(self) -> None:
-        """Setup the dataset."""
-        self.df = self.feature_fn(pd.read_csv(self.filepath, **self.kwargs))
+    def setup_datasource(self) -> None:
+        """Setup the datasource ready to collect statistics."""
+        df = self.feature_fn(pd.read_csv(self.filepath, **self.kwargs))
+        # KeyDataset implementation, sets self.df
+        DataFrameDataset.__init__(self, df)
         self.min_max_scaler.fit(self.df.values)
         self.standardizer.fit(self.df.values)
```

Comment:
I'm sure we could find a way executing only the necessary one by passing the arguments from the _TableDataset. But do you think it's ok to leave it as is in this PR?

Reply:
Surely ok, since this is the eager version it shouldn't matter so much. But in the lazy case of large datasets, where/how is the standardization done? This may create a bottleneck. I'd rather add it into this PR than opening the box again after we merge. But yeah, if you feel like maybe just open a separate issue.